src/richie/apps/courses/models/course.py

```diff
@@ -482,6 +482,32 @@ def course_runs_enrollment_count(self):
             direct_course__extended_object__publisher_is_draft=is_draft,
         ).aggregate(sum=Sum("enrollment_count"))["sum"]
 
+    @cached_property
+    def course_languages_display(self):
+        """
+        Returns a query that languages avaliable of each course run(merged). They may be directly
+        related to the course or to a snapshot of the course.
```

Comment:
```suggestion
        Returns a string listing of languages available for this course across all its course runs (merged).
        They may be directly related to the course or to a snapshot of the course.
```

This second sentence raises a philosophical question : should we list here the languages of archived courses or just open and future courses? 
- yes because they do exist and could always be available in the future again?
- no because you have no way to subscribe to a course in this language atm?

Reply:
My mistake, it was the comment of my first try, I forgot to change the comment after do another changes, sorry about that. I will fix that =)

Reply:
> By reviewing the PR, I have the same question then I checked how we index course languages within ES and currently we do not exclude archived course runs. So from my point of view, if we want to not display archived course runs languages, there is more work to achieve to align search index.

Ok. Yes, we can find courses by language and the archived availability state - https://demo.richie.education/en/courses/?availability=archived&languages=fr&limit=21&offset=0 so it needs to be indexed in the Elastic Search. I think, if a course has different languages where each run has different languages than the other runs, example: run in Mandarin another run in French, but the Mandarin is archived, then we should display it to the user. How about we display it on the same row of the "Upcoming" and "Archived"? Example: _Run 2, from Sept. 7, 2023 to Nov. 15, 2023, Mandarin_?

Can this change be included in the issue of https://github.com/openfun/richie/issues/1991 ? Or at least this new requirement.